python/cudf/cudf/core/index.py

```diff
@@ -2122,6 +2224,18 @@ def _get_dt_field(self, field):
     def is_boolean(self):
         return False
 
+    def _clean_nulls_from_index(self):
+        # Converting to string Index is necessary for DatetimeColumn
+        # because larger values will easily overflow while being
+        # converted to pandas later.
+        # Converting to CategoricalIndex is necessary to maintain repr
+        # formatting, as StringIndex is resulting in drastically different
+        # output.
```

Comment:
Can we go into more detail here? Please feel free to modify before merging this suggestion (probably will fail a style check):

```suggestion
        # __repr__ for other data types works by converting
        # to Pandas first, and relying on Pandas to convert
        #  values to strings. However,
        # Pandas encounters issues with overflow for datetime
        # and timedelta types because it does not support
        # sub-nanosecond resolutions. Thus, we do the work of
        # converting datetimes/timedeltas to strings before handing
        # off to Pandas.
        # 
        # Further, we need to cast the result to a `CategoricalIndex`,
        # because `StringIndex` values in Pandas are printed on
        # the same line, rather than one-per-line.
```